src/legacy/server/kbn_server.js

```diff
@@ -198,9 +197,7 @@ export default class KbnServer {
     return await this.server.inject(opts);
   }
 
-  applyLoggingConfiguration(settings) {
-    const config = new Config(this.config.getSchema(), settings);
-
+  applyLoggingConfiguration(config) {
```

Comment:
This is only called from https://github.com/elastic/kibana/blob/6e6325b1dfe7e75e11a515ec51f638ab5beb4a46/src/core/server/legacy/legacy_service.ts#L108

The only difference between this config object and the one we inject when constructing KbnServer is that this one isn't extended by the legacy plugin discovery. So it won't include any plugin config keys, but as this is used only for logging it shouldn't be a problem.